doc/decisions/deferred_plugin_calls.md

```diff
@@ -0,0 +1,61 @@
+# Deferred Plugin Calls
+
+## Issue
+
+Since some plugins may lazy-load encapsulated plugins at a later time a
+mechanism for deferring function calls to this encapsulated plugins is needed.
+
+For example when setting I/O bindings with `elektraIoSetBinding()` the exported
+function `setIoBinding` is called for all globally mounted plugins.
+The "list" does not export this functions and cannot implement every exported
+function for its encapsulated plugins.
+
+## Constraints
+
+1. Plugins encapsulating other plugins shall be able to lazy-load them.
+2. Callers shall not be entangled in encapsulating plugin details.
+3. The encapsulation of the plugins shall not be broken.
+4. The mechanism shall be generic (not only for I/O bindings).
+
+## Assumptions
+
+1. The called functions do not return a value (e.g. `set`, `open`, `close`, ...)
```

Comment:
Is it possible to communicate in both directions via the KeySet parameter?

Reply:
It is possible but I added that is is not recommended: There is no defined moment where the data should be available and defining such a moment would break lazy-loading.